includes/validation/class-amp-invalid-url-post-type.php

```diff
@@ -0,0 +1,1219 @@
+				break;
+			}
+		}
+	}
+
+	/**
+	 * Gets validation errors for a given invalid URL post.
+	 *
+	 * @param int|WP_Post $post Post of amp_invalid_url type.
+	 * @param array       $args {
+	 *     Args.
+	 *
+	 *     @type bool $ignore_accepted Exclude validation errors that are accepted. Default false.
+	 * }
+	 * @return array List of errors.
+	 */
+	public static function get_invalid_url_validation_errors( $post, $args = array() ) {
+		$args   = array_merge(
+			array(
+				'ignore_accepted' => false,
+			),
+			$args
+		);
+		$post   = get_post( $post );
+		$errors = array();
+
+		$stored_validation_errors = json_decode( $post->post_content, true );
+		if ( ! is_array( $stored_validation_errors ) ) {
+			return array();
+		}
+		foreach ( $stored_validation_errors as $stored_validation_error ) {
+			if ( ! isset( $stored_validation_error['term_slug'] ) ) {
+				continue;
+			}
+			$term = get_term_by( 'slug', $stored_validation_error['term_slug'], AMP_Validation_Error_Taxonomy::TAXONOMY_SLUG );
+			if ( ! $term ) {
+				continue;
+			}
+			if ( $args['ignore_accepted'] && AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACCEPTED_STATUS === $term->term_group ) {
+				continue;
+			}
+			$errors[] = array(
+				'term' => $term,
+				'data' => array_merge(
+					json_decode( $term->description, true ),
+					array(
+						'sources' => $stored_validation_error['sources'],
+					)
+				),
+			);
+		}
+		return $errors;
+	}
+
+	/**
+	 * Get counts for the validation errors associated with a given invalid URL.
+	 *
+	 * @param int|WP_Post $post Post of amp_invalid_url type.
+	 * @return array Term counts.
+	 */
+	public static function get_invalid_url_validation_error_counts( $post ) {
+		$counts = array_fill_keys(
+			array(
+				AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_STATUS,
+				AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_ACCEPTED_STATUS,
+				AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_REJECTED_STATUS,
+			),
+			0
+		);
+
+		$validation_errors = self::get_invalid_url_validation_errors( $post );
+		foreach ( wp_list_pluck( $validation_errors, 'term' ) as $term ) {
+			if ( isset( $counts[ $term->term_group ] ) ) {
+				$counts[ $term->term_group ]++;
+			}
+		}
+		return $counts;
+	}
+
+	/**
+	 * Display summary of the validation error counts for a given post.
+	 *
+	 * @param int|WP_Post $post Post of amp_invalid_url type.
+	 */
+	public static function display_invalid_url_validation_error_counts_summary( $post ) {
+		$result = array();
+		$counts = self::get_invalid_url_validation_error_counts( $post );
+		if ( $counts[ AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_STATUS ] ) {
+			$result[] = esc_html( sprintf(
+				/* translators: %s is count */
+				__( '&#x2753; New: %s', 'amp' ),
+				number_format_i18n( $counts[ AMP_Validation_Error_Taxonomy::VALIDATION_ERROR_NEW_STATUS ] )
```

Comment:
Nice use of [number_format_i18()](https://codex.wordpress.org/Function_Reference/number_format_i18n), I didn't know about that.